### PR TITLE
Update docs for debugging server plugins

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

Old docs
https://developers.mattermost.com/integrate/plugins/developer-workflow/#debug-server-side-plugins-using-delve

New docs
http://mattermost-dev-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/1370/integrate/plugins/developer-workflow/#debug-server-side-plugins-using-delve

This PR updates the section on debugging server-side plugins on the Developer Workflow page. There are some changes due to previously having external libraries checked in a `vendor` folder, which now exist in Go modules.

The PR includes a script to automate modifying the [rpc_client.go](https://github.com/hashicorp/go-plugin/blob/586d14f3dcef1eb42bfb7da4c7af102ec6638668/rpc_client.go#L66) file in the `go-plugin` package, to allow for long pauses from the debugger. Note there is an open PR upstream to implement this into the package https://github.com/hashicorp/go-plugin/pull/101, as well as an issue I had opened for this a while ago https://github.com/hashicorp/go-plugin/issues/248. Patching the library locally for development purposes seems harmless to me. I'm wondering if there's a more appropriate way to do this.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-developer-documentation/issues/1365

### CODE REVIEW

File: site/content/integrate/plugins/developer-workflow.md [hunk 1]
```diff
@@ -50,16 +50,73 @@ If you would like to avoid using ngrok, there is another free option that you ca
...
+
+In the script below, we automatically modify the file located at `${GOPATH}/pkg/mod/github.com/hashicorp/go-plugin@${GO_PLUGIN_PACKAGE_VERSION}/rpc_client.go`, where `GO_PLUGIN_PACKAGE_VERSION` is the version of `go-plugin` that your Mattermost server is using. This can be found in your local copy of the monorepo at [server/go.mod](https://github.com/mattermost/mattermost/blob/4bdd8bb18e47d16f9680905972516526b6fd61d8/server/go.mod#L141).
+
+This script essentially replaces the line in [go-plugin/rpc_client.go](https://github.com/hashicorp/go-plugin/blob/586d14f3dcef1eb42bfb7da4c7af102ec6638668/rpc_client.go#L66) to have a custom configuration for the RPC client connection, that disables the "keep alive" feature. This makes it so the debugger can be paused for long amounts of time, and the Mattermost server will keep the connection with the plugin open.
```
Comment (mickmister) [inline site/content/integrate/plugins/developer-workflow.md:81]: This section is wordy about describing the script. Also the section should probably contain instructions on how to do this manually, like the previous docs had

File: site/content/integrate/plugins/developer-workflow.md [hunk 2]
```diff
@@ -69,18 +126,38 @@ Using the `make attach-headless` command will allow you to use a debugger and st
...
-    sessionConfig.EnableKeepAlive = true
-    sessionConfig.ConnectionWriteTimeout = time.Minute * 5
-    
-    mux, err := yamux.Client(conn, sessionConfig)
-    ```
+```sh
+make attach-headless
+```
+
+This starts a headless `delve` process for your IDE to connect to. The process listens on port `2346`, which is the port defined in our `launch.json` configuration. Somewhat related, the Mattermost server's `Makefile` has a command `debug-server-headless`, which starts a headless `delve` process for the Mattermost server, listening on port `2345`. So you can create a similar `launch.json` configuration in the server directory of the monorepo to connect to your server by using that port.
```
[the remark this answers is not in the document]
Comment (azigler) [reply site/content/integrate/plugins/developer-workflow.md:148]: Are they just missing? What would be needed to add them?

**Review by azigler — APPROVED** [review]
Really fantastic update, @mickmister! 👍 💯  Thanks for taking the time to update this.

**Comment by mickmister** [comment]
> Are they just missing? What would be needed to add them?

@azigler Yes they need to be added to the repos as `.vscode/launch.json` files https://learn.microsoft.com/en-us/microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension/launch-json

**Comment by azigler** [comment]
> > Are they just missing? What would be needed to add them?
> 
> @azigler Yes they need to be added to the repos as `.vscode/launch.json` files https://learn.microsoft.com/en-us/microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension/launch-json

Could we provide an example file inline in this docs PR?

**Review by hanzei — APPROVED** [review]
Awesome :rocket:

**Comment by mickmister** [comment]
> Could we provide an example file inline in this docs PR?

@azigler Yeah there's an example `launch.json` config included in the docs here

https://github.com/mattermost/mattermost-developer-documentation/blob/2c3a4e1916b36a44f39dcc7f7bdd9849e8bf1651/site/content/integrate/plugins/developer-workflow.md?plain=1#L120-L132

**Comment by hanzei** [comment]
@cwarnermm We got two dev approvals; I think that is enough to hand the PR to you.

**Review by cwarnermm — APPROVED** [review]
Fantastic dev docs update!

**Pull request merged**